arviz/tests/external_tests/test_data_numpyro.py

```diff
@@ -227,3 +227,27 @@ def test_inference_data_num_chains(self, predictions_data, chains):
         inference_data = from_numpyro(predictions=predictions, num_chains=chains)
         nchains = inference_data.predictions.dims["chain"]
         assert nchains == chains
+
+    def test_mcmc_with_thinning(self):
+        import numpyro
+        import numpyro.distributions as dist
+        from numpyro.infer import MCMC, NUTS
+
+        x = np.random.normal(10, 3, size=100)
+
+        def model_with_thinning(x):
+            numpyro.sample(
+                "x",
+                dist.Normal(
+                    numpyro.sample("loc", dist.Uniform(0, 20)),
+                    numpyro.sample("scale", dist.Uniform(0, 20)),
+                ),
+                obs=x,
+            )
+
+        nuts_kernel = NUTS(model_with_thinning)
+        mcmc = MCMC(nuts_kernel, num_samples=10, num_warmup=2, thinning=2)
```

Comment:
```suggestion
        mcmc = MCMC(nuts_kernel, num_samples=300, num_warmup=100, thinning=thin)
```

Test with couple of different values and more samples?